app/models/user.rb

```diff
@@ -66,6 +66,8 @@ class User < ApplicationRecord
   has_one :preference, dependent: :destroy
   accepts_nested_attributes_for :preference
 
+  has_many :step_codes, as: :parent, dependent: :destroy
```

Comment:
we should move this into a permit project instead.  A step_code should always belong to some sort of projectItem/project which then has responsibilities about ownership and sharing, etc.